library/std/src/sys/path/unix.rs

```diff
@@ -60,3 +60,14 @@ pub(crate) fn absolute(path: &Path) -> io::Result<PathBuf> {
 
     Ok(normalized)
 }
+
+pub(crate) fn is_absolute(path: &Path) -> bool {
+    if cfg!(target_os = "redox") {
+        // FIXME: Allow Redox prefixes
+        path.has_root() || crate::path::has_redox_scheme(path.as_u8_slice())
+    } else {
+        path.has_root()
+            && (cfg!(any(unix, target_os = "hermit", target_os = "wasi"))
+                || path.prefix().is_some())
+    }
```

Comment:
While you are here, mind cleaning up this block? The last condition is a bit difficult to read as-is.

```rust
    if cfg!(target_os = "redox") {
        // FIXME: Allow Redox prefixes
        path.has_root() || crate::path::has_redox_scheme(path.as_u8_slice())
    } else if cfg!(any(unix, target_os = "hermit", target_os = "wasi")) {
        path.has_root()
    } else {
        path.has_root() && path.prefix().is_some()
    }
```